### PR TITLE
Juno/add java parser (resolved merge conflicts from PR 474)

### PR DESCRIPTION
**Pull request by aceslick911**
Hi,
I've forked from this PR: https://github.com/fkling/astexplorer/pull/474 and merged your latest master in to it so we can merge the java parser finally.

Changes:
> Merged fkling/astexplorer master into juno/add-java-parser (Resolved merge conflicts on package.json and yarn.lock)

All credit goes to @junoatscroll

### CODE REVIEW

File: website/src/parsers/java/java-parser.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  typeProps: new Set(['name']),
+
+  loadParser(callback) {
+    require(['java-parser'], callback);
+  },
+
+  parse(parser, code) {
+    const cst = parser.parse(code);
+
+    const clone = JSON.parse(JSON.stringify(cst));
```
Comment (fkling) [inline website/src/parsers/java/java-parser.js:30]: Thank you for doing this! I still don't know why this line is necessary.... any ideas?

File: website/package.json [hunk 2]
```diff
@@ -72,6 +72,7 @@
     "babylon6": "npm:babylon@^6",
     "babylon7": "npm:@babel/parser@^7",
     "cherow": "^1.6.8",
+    "chevrotain": "^7.0.1",
```
Comment (fkling) [inline website/package.json:75]: If this is installed as a dependency of `java-parser`, this shouldn't really be here...

Comment (aceslick911) [reply website/package.json]: Removed 👍

File: website/src/parsers/java/java-parser.js (hunk 1 above)
Comment (aceslick911) [reply website/src/parsers/java/java-parser.js]: This was possible debug code by the author. I've removed it.

**Comment by aceslick911** [comment]
Removed debug code in java-parser.js
```javascript
  parse(parser, code) {
    const cst = parser.parse(code);

    const clone = JSON.parse(JSON.stringify(cst));

    return clone;
  },
```

-> 

```javascript
  parse(parser, code) {
    return parser.parse(code);
  },
```

**Comment by junoatscroll** [comment]
Thank you for taking the lead on this!
The json stringify/parse was a super hacky way of cleaning up some of the AST data structures that had cycles and were making ASTexplorer unhappy. YMMV. A better approach would be to writer better filters in the parser config.

Unfortunately, I don't have capacity to work further on this at the moment.

**Comment by fkling** [comment]
@junoatscroll 

> The json stringify/parse was a super hacky way of cleaning up some of the AST data structures that had cycles and were making ASTexplorer unhappy.

Are you sure that was the actual problem? Because `JSON.stringify` won't actually work if you have cyclic references... but anyway, this can also be solved by adding "offenders" to `_ignoredProperties`.

I will work on getting a new deployment process up an running first and then do a double check whether I encounter any issues with this parser.

**Pull request merged**